src/mscorlib/shared/System/TimeZoneInfo.Unix.cs

```diff
@@ -932,8 +944,14 @@ internal static TimeSpan GetDateTimeNowUtcOffsetFromUtc(DateTime time, out bool
                 if (!string.IsNullOrEmpty(futureTransitionsPosixFormat))
                 {
                     AdjustmentRule r = TZif_CreateAdjustmentRuleForPosixFormat(futureTransitionsPosixFormat, startTransitionDate, timeZoneBaseUtcOffset);
+
                     if (r != null)
                     {
+                        if (!IsValidAdjustmentRuleOffest(timeZoneBaseUtcOffset, r))
```

Comment:
Nit, if NormalizeAdjustmentRuleOffset handled the "is valid" check then this repeated code would just be one call.